clients/src/main/java/org/apache/kafka/common/log/remote/metadata/storage/RLMMWithTopicStorage.java

```
@@ -254,17 +264,17 @@ public void onStopPartitions(Set<TopicPartition> partitions) {
     }
 
     @Override
-    public void onServerStarted() {
-        initialize();
+    public void onServerStarted(final Optional<String> endpoint) {
+        initialize(endpoint);
     }
 
-    private synchronized void initialize() {
+    private synchronized void initialize(final Optional<String> endpoint) {
         if (!initialized) {
             log.info("Initializing all the clients and resources.");
             //create clients
-            createAdminClient();
-            createProducer();
-            createConsumer();
+            createAdminClient(endpoint);
```

I put a comment in `RemoteLogMetadataManager` to remove Optional wrapper. 
We should throw an error(possibly `IllegalArgumentException`) if endpoint is null as it should always exist for the broker and it is needed to create the clients.